src/components/Modal.vue

```diff
@@ -20,34 +20,57 @@
                  <v-checkbox
                   v-if="head.type === 'boolean'"
                   :label="head.text"
+                  :rules="[() => !!head.value || 'This field is required.']"
+                  required
                   v-model="head.value">
                  </v-checkbox>
                  <v-text-field
                   v-if="head.type === 'date'"
                   :label="head.text"
                   type="date"
+                  :rules="[() => !!head.value || 'This field is required.']"
+                  required
                   v-model="head.value">
                  </v-text-field>
                  <v-text-field
                   v-if="head.type === 'number'"
                   :label="head.text"
+                  :rules="[
+                   () => !!head.value || 'This field is required.',
+                   () => !!head.value && head.value > 0 ||
+                   'Number must be bigger than 0.',
+                  ]"
+                  required
                   type="number"
                   v-model="head.value" >
                  </v-text-field>
+                 <v-text-field
+                  v-if="head.type === 'int-number'"
```

Comment:
New modal type for number fields such as latitude and longitude from Sites which can take on negative values.